File: docs/lib/filters/credential_permission_filters.js
import { CatalogueFilters } from "../modules/catalogue.js";

class CredentialPermissionFilters extends CatalogueFilters {

    /** @type {boolean} */
    clean;

    /** @type {string} */
    credentialId;

    /** @type {string} */
    permissionId;

    constructor(json) {
        super(json);
        this.clean = json?.clean ?? true;
        this.credentialId = json?.credentialId ?? null;
        this.permissionId = json?.permissionId ?? null;
    }

}

export {
    CredentialPermissionFilters
}

File: Core/Filters/CredentialPermissionFilters.cs
using KolibSoft.Catalogue.Core;

namespace KolibSoft.AuthStore.Core.Filters;

public class CredentialPermissionFilters : CatalogueFilters
{
    public bool? Clean { get; init; }
    public Guid? CredentialId { get; init; }
    public Guid? PermissionId { get; init; }
}